### PR TITLE
Add failing test for @content-helper lookup

### PR DESCRIPTION
**Pull request by tim-evans**
Related to https://github.com/ember-cli/ember-resolver/issues/101

### CODE REVIEW

**Comment by rwjblue** [comment]
Ok. Now fix it ;)

**Comment by rwjblue** [comment]
Looks good to me.

@stefanpenner - Thoughts?

**Comment by stefanpenner** [comment]
can someone describe concisely the use of `@` here, and how it relates to the existing namespace usage? In this case is it merely referring to the current namespace?

**Comment by rwjblue** [comment]
@stefanpenner - This `helper:@content-helper` lookup is done by HTMLBars (I assume that it uses an `@` to prevent collisions or something), so the `@` here has nothing to do with the namespace concept that we have added to this resolver (which is why we are trying to detect invalid usage and falling back out of the "namespaced lookup" path).

**Comment by rwjblue** [comment]
I wrote a little more of a rambling explanation in https://github.com/ember-cli/ember-resolver/issues/101#issuecomment-128072663 and https://github.com/emberjs/ember.js/issues/11987.

**Comment by stefanpenner** [comment]
@rwjblue thanks for the links, let me read the backchat.

File: packages/ember-resolver/lib/core.js [hunk 1]
```diff
@@ -57,6 +57,10 @@ define("ember/resolver",
     var prefix, type, name;
     var fullNameParts = fullName.split('@');
 
+    if (/.*:@.*/.test(fullName)) {
```
Comment (stefanpenner) [inline packages/ember-resolver/lib/core.js]: * lets leave a comment here describing why this is needed
* it seems like the above split `fullName.split('@');` (on line 58) is not needed if we hit this branch, this should be cleaned up

**Comment by tim-evans** [comment]
@stefanpenner Is the comment and changes I made sufficient?

**Comment by stefanpenner** [comment]
@tim-evans yup, LG.

@rwjblue you are more familiar with that part of HTMLBars, so merge & release when you are comfortable with it.

**Comment by rwjblue** [comment]
OK, I did a bit more digging into what was happening and I'd prefer to specifically blacklist `helper:@content-helper` for now.  It is a "magic" value (that @mmun is going to try and remove in HTMLBars), and the issue should not occur on canary already (because of cleanup work done to remove support for `Ember.Handlebars.makeViewHelper`).

@tim-evans - Can you tweak (hopefully for the last time) to just compare `fullName === 'helper:@content-helper'`?

**Comment by mmun** [comment]
Can't this change be made in htmlbars instead?

**Comment by tim-evans** [comment]
@rwjblue done.

**Comment by rwjblue** [comment]
Beautiful!

**Comment by mmun** [comment]
Ok rwjblue convinced me this is good

**Comment by tim-evans** [comment]
Could we get a release on this? ember-document-title doesn't work on 1.13.X because of this. :(

I checked on 1.13.8 and it no longer throws errors when calling `{{title}}`, but calls the helper, which is right at all.

**Comment by stefanpenner** [comment]
releasing now.

**Comment by stefanpenner** [comment]
```
➜  ember-resolver git:(master) git tag v0.1.19
➜  ember-resolver git:(master) git push origin v0.1.19
Total 0 (delta 0), reused 0 (delta 0)
To git@github.com:ember-cli/ember-resolver
```

**Comment by stefanpenner** [comment]
@tim-evans let me know if that does the trick, sorry for letting this linger.

**Comment by tim-evans** [comment]
Sadly, no. I'll come up with an ember-twiddle to report this.

**Comment by stefanpenner** [comment]
@tim-evans ah so an additional bug in the resolver?

**Comment by tim-evans** [comment]
http://ember-twiddle.com/b3b796e77b72d2e6c816

**Comment by tim-evans** [comment]
I'm not sure.

**Comment by stefanpenner** [comment]
@tim-evans can you open a new issue, describing what is expected. I would love to dive in later this morning :)

**Pull request merged**